tools/update_tool

```diff
@@ -0,0 +1,466 @@
+                    if "fstype" in partition:
+                        if partition["fstype"] == "iso9660":
+                            partition["uuid"] = "2022-07-01-fixed-uuid"
+        return imi
+    return iso(lvm2(imi))
+
+
+def diff(what, json1, json2, tool="vimdiff"):
+    """
+    Starts a diff between two json objects with a specific tool to do so. Writes
+    each json to disk in a temporary directory and then invokes the difftool on
+    these two files.
+    """
+    with tempfile.TemporaryDirectory() as tempdir:
+        a = os.path.join(tempdir, "1")
+        b = os.path.join(tempdir, "2")
+        with open(a, encoding="utf-8", mode="w") as f:
+            json.dump(json1, f, indent=4)
+        with open(b, encoding="utf-8", mode="w") as f:
+            json.dump(json2, f, indent=4)
+        opendiff = input(f"-   diff {what}? [Y, n]")
+        if opendiff in ("y", "Y", "", None):
+            subprocess.call([tool,  a, b])
+
+
+def extract(db_entry, do_filter=True):
+    """
+    Returns two json objects from a db entry. One manifest and one image info.
+    Can apply (and will by default) filters to the image-info to get rid of some
+    volatile UUIds that aren't important.
+    """
+    manifest = None
+    imi = None
+    if not db_entry:
+        return None, None
+    if db_entry["image-info"]:
+        if do_filter:
+            imi = filter_fn(db_entry["image-info"])
+        else:
+            imi = db_entry["image-info"]
+    if db_entry["manifest"]:
+        manifest = db_entry["manifest"]
+    return manifest, imi
+
+
+def get_symbol(label):
+    """
+    Get an icon to sum up what changed
+    """
+    if label == "added":
+        return "🆕"
+    if label == "updated":
+        return "🔁"
+    if label == "removed":
+        return "🗑️"
+    if label == "unchanged":
+        return "💾"
+    return None
+
+
+
+
+def progress(count, total, prefix=''):
+    """
+    Prints a nice progress bar that takes up to the all width of the terminal
+    """
+    bar_len = os.get_terminal_size().columns - 10 - len(prefix)
+    filled_len = int(round(bar_len * count / float(total)))
+
+    percents = round(100.0 * count / float(total), 1)
+    barr = '-' * (filled_len-1) + ">" + ' ' * (bar_len - filled_len)
+
+    sys.stdout.write(f'{prefix}{percents}%[{barr}]\r')
+    sys.stdout.flush()
+
+
+def erase_bar():
+    """
+    Erases the progress bar
+    """
+    sys.stdout.write(f'{" "*os.get_terminal_size().columns}\r')
+    sys.stdout.flush()
+
+
+def ls_command(args):
+    """
+    Lists all the changed files between this and the previous commit and puts
+    some icons for the user to know what changed and what not
+    """
+    files = []
+    lst = list_changed_files()
+    for i, file in enumerate(lst):
+        filename = file.split("/")[-1].split(".")[0]
+        prev = get_db_entry_at_previous_version(file)
+        curr = get_db_entry_at_current_version(file)
+
+        if not (prev or curr):
+            continue
```

Comment:
If one of the two doesn't exist, couldn't we just shortcut the function to say something was introduced (if prev is None) or something was removed (if curr is None)?

Reply:
Thanks for review! Yes that's a good idea. I think the logic is already kind of handled, because when you are in an added or removed state you'll get the info. Maybe you'd rather want the code to avoid opening you a diff in these two cases ?
At the beginning I wanted to only open the diff tool on the updated state.

We could instead of having:

```
        if manifest_status != "unchanged":
            diff("manifest", prev_manifest, curr_manifest, tool=args.manifest_diff_tool)

        if image_info_status != "unchanged":
            diff("image-info", prev_imi, curr_imi, tool=args.image_info_diff_tool)
```
            
have:
      
```
        if manifest_status == "updated":
            diff("manifest", prev_manifest, curr_manifest, tool=args.manifest_diff_tool)

        if image_info_status == "updated":
            diff("image-info", prev_imi, curr_imi, tool=args.image_info_diff_tool)
``` 
 
 What do you think ?

Reply:
@achilleas-k, I'll mark this as resolved, but feel free to reopen it.